### PR TITLE
[Job-exporter] Fix gpu matrix not match task-role issue

### PR DESCRIPTION
**Pull request by Binyang2014**
Tested in vnext bed. After this fix, the task role metrics show correctly
Refer: https://github.com/NVIDIA/nvidia-docker/issues/376
GPU minor number not match NVIDIA_VISIBLE_DEVICES  sometimes

### CODE REVIEW

File: src/job-exporter/src/nvidia.py [hunk 1]
```diff
@@ -101,8 +101,13 @@ def parse_smi_xml_result(smi):
...
 
-    for gpu in gpus:
-        minor = gpu.getElementsByTagName("minor_number")[0].childNodes[0].data
+    for index, gpu in enumerate(gpus):
+        if os.getenv("LAUNCHER_TYPE") == "k8s":
+            # For pai k8s, the minor number doesn't match the NVIDIA_VISIBLE_DEVICES number,
+            # use nvidia-smi gpu sequence index instead
+            minor = index
+        else:
+            minor = gpu.getElementsByTagName("minor_number")[0].childNodes[0].data
```
Comment (abuccts) [inline src/job-exporter/src/nvidia.py]: change `minor` to another name, like `gpu_index`?

Comment (Binyang2014) [reply src/job-exporter/src/nvidia.py]: fixed

**Comment by xudifsd** [comment]
suggest adding test cases for this kind of change

**Comment by Binyang2014** [comment]
> suggest adding test cases for this kind of change

Fixed

**Pull request merged**